src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,11 +31,77 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
+        int numberOfParts;
+        this.value = new String(trimmedAddress);
```

Comment:
Why is the `value` still being retained? Consider all the mishaps that can happen due to data duplication!

Also, consider moving the validation into the new classes you created.